sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -1431,6 +1415,67 @@ void Scheduler::GraphBuilder::cancelFusion(QueueImplPtr Queue,
   PlaceholderCmd->setFusionStatus(KernelFusionCommand::FusionStatus::CANCELLED);
 }
 
+static bool isPartOfFusion(Command *Cmd, KernelFusionCommand *Fusion) {
+  if (Cmd->getType() == Command::RUN_CG) {
+    return static_cast<ExecCGCommand *>(Cmd)->MFusionCmd == Fusion;
+  }
+  return false;
+}
+
+static bool dependsOnFusion(Command *, bool, KernelFusionCommand *);
```

Comment:
```suggestion
static bool checkForCircularDependency(Command *, bool, KernelFusionCommand *);
```

`dependsOnFusion` seems highly specific to circular dependency.